independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/matching/RegexPathTest.java

```diff
@@ -0,0 +1,114 @@
+package org.jboss.resteasy.reactive.server.vertx.test.matching;
+
+import static io.restassured.RestAssured.get;
+import static org.hamcrest.Matchers.equalTo;
+
+import java.util.function.Supplier;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+import org.jboss.resteasy.reactive.server.vertx.test.framework.ResteasyReactiveUnitTest;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class RegexPathTest {
+
+    @RegisterExtension
+    static ResteasyReactiveUnitTest test = new ResteasyReactiveUnitTest()
+            .setArchiveProducer(new Supplier<>() {
+                @Override
+                public JavaArchive get() {
+                    return ShrinkWrap.create(JavaArchive.class)
+                            .addClasses(EndingSlashTest.TestResource.class);
+                }
+            });
+
+    @Test
+    public void test() {
+
+        get("/hello/world/1")
+                .then()
+                .statusCode(200)
+                .body(equalTo("Hello World! 1"));
+
+        get("/hello/world/1/")
+                .then()
+                .statusCode(200)
+                .body(equalTo("Hello World! 1"));
+
+        get("/hello/")
+                .then()
+                .statusCode(404);
+
+        get("/hello/1")
+                .then()
+                .statusCode(200)
+                .body(equalTo("Hello! 1"));
+
+        get("/hello/1/")
```

Comment:
What happens if you do get `/hello/1////`?

Reply:
Intermediate slashes are encoded, for instance the path becomes `/1%2F%2F%2F%2F/`, which doesn't match the regex path matcher `\\d+`.

Reply:
So, we only alow a single optional slash at the end, then?

Reply:
Yes, that's correct.

Reply:
I'm not sure how I feel about this.
It feels special-casy.

Also, I looked for articles about tailing slashes, and I only find SEO articles that advise to only serve one variant (with, or without).

And last, I've come across this issue https://github.com/spring-projects/spring-framework/issues/28552 where Sprint went the other way around and removed the default of allowing optional trailing slashes by default.

So, I'll ask for more comments from HTTP/REST colleagues who might have a relevant opinion: @cescoffier @vietj:

Do you think we should allow trailing slashes for routes defined without one?

Now, on the other hand, this issue is about fixing the particular difference of assuming a trailing slash for regex paths, so that it's coherent with normal paths, which already assume a trailing slash. And I agree both should be the same. I'm not entirely sure regex matches are right, though, I wonder if non-regex path matching might be in the wrong here and the one that needs fixing. WDYT @geoand ?

Reply:
> And I agree both should be the same. I'm not entirely sure regex matches are right, though, I wonder if non-regex path matching might be in the wrong here and the one that needs fixing. WDYT @geoand ?

I really have not looked at it at all, so I trust you completely on this one

Reply:
/foo and /foo/ are not equivalent. They refer to different resources due to the trailing slash:

* /foo refers to a resource.
* /foo/ often suggests that it’s a collection or directory-like structure containing resources.

Reply:
So @cescoffier you're saying we should change the current behaviour of Quarkus REST to not treat these as the same when routing?

I suppose making the change and running the TCK will tell us what JAX-RS thinks about that :)

Reply:
That's the problem - many servers consider them equivalent, but the "theory" says they are not.

Reply:
There was a similar conversation here https://github.com/quarkusio/quarkus/issues/26016#issuecomment-1152953876

Reply:
Well, OK, so the TCK did not test this, and the spec says we have to treat them equivalent. Probably people rely on that now, so let's just harmonize and make sure regexes also behave in a similar way, and if we want to change the default later, we'll wait for users reporting an issue and revisit if required, for both regular and regex.